src/SDL/Audio.hs

```diff
@@ -206,6 +207,60 @@ getAudioDeviceNames usage = liftIO $ do
 
   where usage' = encodeUsage usage
 
+data AudioFormat = AudioFormat NumberFormat SampleBitSize Endianess
+  deriving (Eq, Ord, Read, Show, Typeable)
+
+data NumberFormat = SignedInteger | UnsignedInteger | Float
+  deriving (Eq, Ord, Read, Show, Typeable)
+
+type SampleBitSize = Word8
+
+data Endianess = LittleEndian | BigEndian | Native
+  deriving (Eq, Ord, Read, Show, Typeable)
+
+encodeAudioFormat :: AudioFormat -> Word16
+encodeAudioFormat (AudioFormat number bits endian) =
+  numberFormat .|. sampleSize .|. endianess
+  where
+    numberFormat =
+      case number of
+        UnsignedInteger -> 0b0000000000000000
+        SignedInteger   -> 0b1000000000000000
+        Float           -> 0b0000000100000000
+
+    sampleSize = fromIntegral bits
+
+    endianess =
+      case endian of
+        BigEndian    -> 0b0001000000000000
+        LittleEndian -> 0b0000000000000000
+        -- Use SDL_endian.h to detect endianess
+        -- https://wiki.libsdl.org/CategoryEndian
+        Native       -> error "not implemented"
```

Comment:
I wouldn't be happy merging with this, perhaps now we should just drop `Native` from `Endianness` and implement that later?